libvmi/driver/xen/xen_events.c

```diff
@@ -1883,6 +1888,9 @@ status_t process_requests_412(vmi_instance_t vmi, uint32_t *requests_processed)
         rsp->reason = vmec.reason;
         rsp->altp2m_idx = vmec.altp2m_idx;
 
+        if (rsp->flags & VM_EVENT_FLAG_FAST_SINGLESTEP)
+          rsp->u.fast_singlestep.p2midx = vmec.fast_singlestep.p2midx;
```

Comment:
This will also likely will need to move to the new functions that will handle the Xen 4.14 vm_event ABI ( VM_EVENT_INTERFACE_VERSION 0x00000006).

Reply:
I've added the plumbing for vm_event version 6 so this addition should be a lot easier.

Reply:
Thank you!